src/bptl/work_units/email/tasks.py

```diff
@@ -0,0 +1,89 @@
+from django.conf import settings
+from django.core.mail import send_mail
+from django.template.loader import get_template
+from django.utils.translation import ugettext_lazy as _
+
+from zgw_consumers.api_models.base import factory
+
+from bptl.tasks.base import WorkUnit
+from bptl.tasks.registry import register
+
+from .serializers import SendEmailSerializer, ValidTemplateChoices
+
+__all__ = ["SendEmail"]
+
+
+@register
+class SendEmailTask(WorkUnit):
+    """This task sends an email to ``receiver`` signed by ``sender``.
+
+    **Required process variables**
+
+    * ``sender``: JSON with email and name of sender.
+        .. code-block:: json
+                {
+                    "email": "kees@example.com",
+                    "name": "Kees Koos"
+                }
+
+    * ``receiver``: JSON with email and name of receiver.
+        .. code-block:: json
+                {
+                    "email": "jan@example.com",
+                    "name": "Jan Janssen"
+                }
+
+    * ``email``: JSON with email variables:
+        .. code-block:: json
+                {
+                    "subject": "This is an example subject.",
+                    "content": "This is an example body."
+                }
+
+    * ``template``: string with template name. Choices:
+        .. code-block:: list
+                [
+                    generiek, accordering, advies,
+                ]
+
+    **Optional process variables**
+
+    * ``kownslFrontendUrl``: string with full URL for end users to submit their review.
+                Valid choices are currently: generic/approval/advice.
+    * ``reminder``: boolean.
+    * ``deadline``: datetime: "2020-04-20 16:20:00".
+    """
+
+    def perform(self):
+        variables = self.task.get_variables()
+        send_email = SendEmailSerializer(data=variables)
+        send_email.is_valid(raise_exception=True)
```

Comment:
I like the usage of using a serializer for validation, but now the errors will look different from what people are using in the Camunda incident logging and the BPTL exception details.

Can you somehow turn this into `bptl.tasks.base.MissingVariable` exceptions where relevant? This is why `bptl.tasks.base.check_variable` exists.

```suggestion
        send_email = SendEmailSerializer(data=variables)
        send_email.is_valid(raise_exception=True)
```

Reply:
absolutely. had something with check_variable first but I like serializers better... will find a middle ground solution

Reply:
cool! it did give me the idea that we should probably layer our own validation on top of DRF serializers, but I don't expect this to be done before 1.0 - could be a 1.1 refactor thing and I'd appreciate input!

Reply:
Right now I did an error catch and I'm matching the error_code to raise a MissingVariable error where relevant. It's definitely not ideal. 

In this case we want our validation to run before the DRF serializers validation (and its fields' validation), so custom (field/serializer) validators aren't useful because they get called much later on.

I was thinking in the direction of writing a custom serializer object with some class methods to check for required fields and recursively looking for the fields in the initial_data. Then we could have the relevant serializers inherit the is_valid function from our custom serializer. Would be nice to hear your arguments against. :-)

Reply:
I'm having a hard time picturing what you mean, can you make a gist with some small (pseudo) code examples to illustrate it? We can discuss on Slack then. There's also no urgency for this, so don't feel pressured to prio this!